File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-6.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        d={}
        l=0
        maxe=res=0
        for r,i in enumerate (s):
            d[i]=1+d.get(i,0)
            maxe=max(maxe,d[i])

            while (r-l+1) - maxe >k:
                d[s[l]]-=1
                l+=1
            res=max(res,r-l+1)
        return res



        

File: Data Structures & Algorithms/trapping-rain-water/submission-6.java
class Solution {
    public int trap(int[] height) {
        int leftmax = height[0];
        int n= height.length;
        int rightmax = height[n-1];
        int l=0;
        int r = n-1;
        int res =0;
        while (l<r){
            if (leftmax < rightmax){
                l++;
                leftmax = Math.max(leftmax,height[l]);
                res += leftmax - height[l];
            }
            else{
                r--;
                rightmax = Math.max(rightmax,height[r]);
                res += rightmax - height[r];
            }

        }
        return res;

        
    }
}
